src/System.Private.CoreLib/shared/System/MemoryExtensions.cs

```diff
@@ -22,6 +22,424 @@ namespace System
+            where T : IEquatable<T>
+        {
+            (int start, int length) = TrimHelper(in span, trimElement, false, true);
+            return span.Slice(start, length);
+        }
+
+        /// <summary>
+        /// Conditionally removes all leading and trailing occurrences of a specified element.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="span">The source span from which the element is removed.</param>
+        /// <param name="trimElement">The specified element to look for and remove.</param>
+        /// <param name="trimStart">If set to True, trims the start.</param>
+        /// <param name="trimEnd">If set to True, trims the end.</param>
+        private static (int start, int length) TrimHelper<T>(in ReadOnlySpan<T> span, in T trimElement, bool trimStart, bool trimEnd)
+            where T : IEquatable<T>
+        {
+            if (span.IsEmpty)
+                return (0, 0);
+
+            int start = 0;
+            int end = span.Length - 1;
+
+            if (trimElement == null)
+            {
+                if (trimStart)
+                {
+                    for (; start < span.Length; start++)
+                    {
+                        if (span[start] != null)
+                            break;
+                    }
+                }
+
+                if (trimEnd)
+                {
+                    // Stop searching once we reach start
+                    for (; end >= start; end--)
+                    {
+                        if (span[end] != null)
+                            break;
+                    }
+                }
+            }
+            else
+            {
+                if (trimStart)
+                {
+                    for (; start < span.Length; start++)
+                    {
+                        if (span[start] != null && !span[start].Equals(trimElement))
+                            break;
+                    }
+                }
+
+                if (trimEnd)
+                {
+                    // Stop searching once we reach start
+                    for (; end >= start; end--)
+                    {
+                        if (span[end] != null && !span[end].Equals(trimElement))
+                            break;
+                    }
+                }
+            }
+
+            return (start, end - start + 1);
+        }
+
+        /// <summary>
+        /// Removes all leading and trailing occurrences of a set of elements specified
+        /// in a readonly span from the memory.
+        /// </summary>
+        /// <param name="trimElements">The span which contains the set of elements to remove.</param>
+        /// <remarks>If <paramref name="trimElements"/> is empty, the memory is returned unaltered.</remarks>
+        public static Memory<T> Trim<T>(this Memory<T> memory, ReadOnlySpan<T> trimElements)
```

Comment:
Perf: Should either of these public parameters be decorated as `in` ?

Reply:
No